.tekton/determine-image-tag-task-oci-ta.yaml

```diff
@@ -41,4 +41,8 @@ spec:
       dnf -y install git make
 
       .konflux/scripts/fail-build-if-git-is-dirty.sh
+
+      BUILD_TAG="$(git describe --exact-match --tags HEAD)" || echo "Warning: Cannot get tag"
+      export BUILD_TAG
```

Comment:
This way it works for the stackrox repo. However, neither scanner nor collector understand `BUILD_TAG` environment variable. For Scanner it is `RELEASE_TAG`, for Collector it is `COLLECTOR_TAG`.

Why does this matter? Because the code here exports `BUILD_TAG` environment variable and expects that `make ... tag` will bring it back. If we copy-paste `determine-image-tag-task-oci-ta.yaml` code across repos, the magic won't happen because Collector and Scanner don't recognize `BUILD_TAG`.

I know that `determine-image-tag-task-oci-ta.yaml` already differs between the repos and it may keep diverging in the future. We, however, should try not make it worse.

Another argument I have is that environment variables are sneaky and this is a perfect example. If you copy-paste this code to Scanner or Collector repos, nothing will fail telling you that `BUILD_TAG` has no effect or that `RELEASE_TAG`/`COLLECTOR_TAG` was expected but not provided.
Environment variables are essentially globals and globals are well-known for not being a developer-friendly tool for passing things around. We definitely do have abuse of environment variables in our scripting.

My suggestion is pretty simple: if `determine-image-tag-task-oci-ta.yaml` detects that the commit is tagged, it should return that git tag without calling into `make tag`.

Whether or not to apply `-fast` suffix to the tag is a good question, let's discuss it in your [Slack thread](https://redhat-internal.slack.com/archives/C05TS9N0S7L/p1730206231947449).

After `determine-image-tag-task-oci-ta.yaml` is isolated from environment variables, there's still need for `make ... tag` to produce the same result during _container builds_. I think this should be already accounted for in all container builds because we pass `$(tasks.determine-image-tag.results.IMAGE_TAG)` in `BUILD_ARGS` and further the arg gets passed to a proper environment variable.

Reply:
> My suggestion is pretty simple: if determine-image-tag-task-oci-ta.yaml detects that the commit is tagged, it should return that git tag without calling into make tag.

It should work like that. The "hard part" was coming up with how to detect tagged commits (luckily we already had this command in our main repo). 

> I think this should be already accounted for in all container builds because we pass $(tasks.determine-image-tag.results.IMAGE_TAG) in BUILD_ARGS and further the arg gets passed to a proper environment variable.

I think so too: 
* https://github.com/stackrox/stackrox/blob/master/.tekton/main-pipeline.yaml#L239-L242
* https://github.com/stackrox/stackrox/blob/master/image/rhel/konflux.Dockerfile#L123-L124

Reply:
`git describe --exact-match --tags HEAD` has its limitation though. Try

```
$ git checkout 3.70.0
$ git describe --exact-match --tags HEAD
$ git tag --points-at
```

Reply:
Update: git tag sorts lexicographically by default.
On an artificial repo:
```bash
$ git tag --points-at
20.6.0
3.70.0
3.70.0-rc.6
3.74.0

$ git describe --exact-match --tags HEAD
20.6.0

$  git tag --sort=version:refname --points-at
3.70.0
3.70.0-rc.6
3.74.0
20.6.0

$ git tag --points-at | sort --version-sort
3.70.0
3.70.0-rc.6
3.74.0
20.6.0
```

I don't know if `--sort=version:refname` is universally supported; perhaps `| sort --version-sort` could be more portable.

Reply:
In the current release automation, there is no way that a commit is tagged multiple times. 
Each new RC or final release is a new empty commit. This comes from when we used to build and test on OSCI, now it is still present to trigger new OSCI tests, as OSCI only triggers on new commits. 

As long as we use GHA+OSCI for upstream and as 'daily driver', we should be safe.

My suggestion would be to add a check that the current commit is only tagged once (as easy as `git tag --points-at | wc -l == 1`), document this assumption in one of the fast stream/daily driver tickets for Konflux and find a solution later.

Reply:
I like the suggestion: fail fast instead of making potentially wrong assumption.
I have https://github.com/stackrox/collector/pull/1920 where I modified `determine-image-tag-task-oci-ta.yaml`; will use this approach there.